pkg/virt-controller/watch/vm.go

```diff
@@ -664,6 +665,108 @@ func (c *VMController) handleCPUChangeRequest(vm *virtv1.VirtualMachine, vmi *vi
 	return nil
 }
 
+func (c *VMController) VMINodeSelectorPatch(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) error {
+	var ops []string
+
+	if vm.Spec.Template.Spec.NodeSelector != nil {
+		vmNodeSelector := make(map[string]string)
+		// copy the node selector map
+		for k, v := range vm.Spec.Template.Spec.NodeSelector {
+			vmNodeSelector[k] = v
+		}
+		vmNodeSelectorJson, err := json.Marshal(vmNodeSelector)
+		if err != nil {
+			return err
+		}
+		if vmi.Spec.NodeSelector == nil {
```

Comment:
i've tried to use the PatchOperaton. It works for simple values but doesn't work for the node selector. 
I don't see a reason to hold this up because of this. I'll follow up once I'll figure this out.

Reply:
ACK apologies this was just a nit.